From b92e6c21642dd200e3e352997f20ab9bfb556a61 Mon Sep 17 00:00:00 2001
From: Jim Meyering <meyering@redhat.com>
Date: Tue, 21 Sep 2010 14:33:41 +0000
Subject: build: avoid compilation failure on the Hurd

* src/dfasearch.c (dfawarn): Rename enum symbols to use DW_ prefix,
so as not to collide with "GNU", which is defined by the Hurd.
Reported by Matthias Lanzinger in http://savannah.gnu.org/bugs/?31096
---
diff --git a/src/dfasearch.c b/src/dfasearch.c
index 44eb08a..780891f 100644
--- a/src/dfasearch.c
+++ b/src/dfasearch.c
@@ -61,10 +61,10 @@ dfaerror (char const *mesg)
 void
 dfawarn (char const *mesg)
 {
-  static enum { NONE = 0, POSIX, GNU } mode;
-  if (mode == NONE)
-    mode = (getenv ("POSIXLY_CORRECT") ? POSIX : GNU);
-  if (mode == GNU)
+  static enum { DW_NONE = 0, DW_POSIX, DW_GNU } mode;
+  if (mode == DW_NONE)
+    mode = (getenv ("POSIXLY_CORRECT") ? DW_POSIX : DW_GNU);
+  if (mode == DW_GNU)
     dfaerror (mesg);
 }
 
--
cgit v0.8.3.4
